src/transformers/utils/fx.py

```diff
@@ -318,11 +397,290 @@ def create_arg(self, a: Any) -> Argument:
         return super().create_arg(a)
 
 
+def _insert_batch_size_node(gm: GraphModule) -> Tuple[GraphModule, Node]:
+    """Helper function that inserts a node that retrieves the batch size dynamically from the input of the model."""
+    graph = gm.graph
+    input_names = gm.dummy_inputs.keys()
+    batch_size_node = None
+    for node in graph.nodes:
+        if node.op == "placeholder" and node.name in input_names:
+            with graph.inserting_after(node):
+                batch_size_node = graph.call_method("size", args=(node, 0))
+
+    if batch_size_node is None:
+        raise ValueError("Could not insert the node that computes the batch size")
+
+    graph.lint()
+    gm.recompile()
+
+    return gm, batch_size_node
+
+
+def _insert_encoder_sequence_length_node(gm: GraphModule) -> Tuple[GraphModule, Node]:
+    """
+    Helper function that inserts a node that retrieves the encoder sequence length dynamically from the input of the
+    model.
+    """
+    graph = gm.graph
+    input_names = gm.dummy_inputs.keys()
+    encoder_sequence_length_node = None
+    for node in graph.nodes:
+        if (node.op == "placeholder") and ("decoder" not in node.name) and (node.name in input_names):
```

Comment:
probably dropping parentheses for consistency with the rest of the code? Did the autoformatter mess it up?